lua/astronvim/utils/ui.lua

```diff
@@ -200,16 +203,20 @@ function M.toggle_wrap()
 end
 
 --- Toggle syntax highlighting and treesitter
-function M.toggle_syntax()
+---@param bufnr? number the buffer to toggle syntax on
+function M.toggle_syntax(bufnr)
   local ts_avail, parsers = pcall(require, "nvim-treesitter.parsers")
-  if vim.g.syntax_on then -- global var for on//off
-    if ts_avail and parsers.has_parser() then vim.cmd.TSBufDisable "highlight" end
-    vim.cmd.syntax "off" -- set vim.g.syntax_on = false
+  bufnr = bufnr or vim.api.nvim_win_get_buf(0)
```

Comment:
```suggestion
  bufnr = bufnr or 0
```

All of these functions/options handle 0 already. No need for us to expand it

Reply:
also @Subjective I made a PR to neovim core to fix this. We do have to expand because of a bug in `vim.treesitter.stop` not aligning APIs: https://github.com/neovim/neovim/pull/24450